### ISSUE TITLE
After the object detection model is trained, how to customize the criteria of ‘best model’?

### ISSUE THREAD

**Issue by 0x6f6f** [ultralytics/ultralytics#12908]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Currently, the default standard used for verification seems to be map50, but sometimes we might need the highest recall rate, or consider both. Is there a way to achieve this?

Say, we got this curve during our training and want to save a better model with higher mar, but Ultralytics v8 saved the model with highest map50.

![maybe i want to save the model with better mar instead of better map](https://github.com/ultralytics/ultralytics/assets/129521056/abfcebdb-94e6-4a26-bfc9-2b55be06ddf3)

 For example, by passing a custom metric function or related configuration options to the framework during training?

**Comment by glenn-jocher** [comment]
Hello! Great question on customizing the criteria for the 'best model' during training with YOLOv8.

Currently, YOLOv8 automatically saves the model with the highest mAP @0x6f6f.5 during training. If you need to prioritize a different metric, such as recall or mAR, you would need to manually monitor these metrics during training and save the model checkpoints accordingly.

While YOLOv8 does not directly support custom metric functions for model saving out-of-the-box, you can implement a simple workaround by modifying the training loop. Here’s a basic example of how you might adjust the training script to save models based on a custom metric:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.yaml')

# Define your custom metric threshold
custom_metric_threshold = 0.7

# Train the model
for epoch in range(100):
    results = model.train(data='coco128.yaml', epochs=1)
    if results.metrics['mar'] > custom_metric_threshold:
        model.save(f'model_at_epoch_{epoch}.pt')
```

This script will save a model checkpoint every time the mAR exceeds your defined threshold during the training epochs.

For more detailed control, you might need to dive deeper into the training function or consider contributing to the repository to add this feature. Don't hesitate to reach out if you need further assistance! 🚀

**Comment by glenn-jocher** [comment]
You're welcome! I'm glad you found the workaround helpful. If you have any more questions or need further assistance as you implement this, feel free to ask. Happy coding! 🚀
